pulp_ansible/app/models.py

```diff
@@ -232,6 +236,65 @@ class Meta:
         unique_together = ("pubkey_fingerprint", "signed_collection")
 
 
+class AnsibleNamespace(Content):
+    """
+    A content type representing the Namespace metadata of a Collection.
+
+    Namespace metadata can have an avatar which is stored as an associated artifact upon upload.
+
+    Fields:
+        name (models.CharField): The required name of the Namespace
+        company (models.CharField): Optional namespace owner company name.
+        email (models.CharField): Optional namespace contact email.
+        description (models.CharField): Namespace brief description.
+        resources (models.TextField): Namespace resources page in markdown format.
+        links (psql_fields.HStore): Labeled related links.
+        avatar_sha256 (models.CharField): SHA256 digest of avatar image.
+        metadata_sha256(models.CharField): SHA256 digest of all other metadata fields.
+    """
+
+    TYPE = "namespace"
+    repo_key_fields = ("name",)
+    # fields on the existing model in galaxy_ng
+    name = models.CharField(max_length=64, blank=False)
+    company = models.CharField(max_length=64, blank=True, default="")
+    email = models.CharField(max_length=256, blank=True, default="")
+    description = models.CharField(max_length=256, blank=True, default="")
+    resources = models.TextField(blank=True, default="")
+
+    links = psql_fields.HStoreField(default=dict)
```

Comment:
Couple reasons: 1. The uniqueness constraint is the sha256 of all fields combine, using one field instead of another table makes it easier to calculate. 2. HStoreField does what the old link table did but faster and easier with Django's queryset API. (We just recently switched to using this field for pulp_labels because we found that the old model we were using had serious performance issues.)

Reply:
I'm wondering which one is the clearer solution:
1. Depend on the pulpcore migration to activate the hstore extension.
2. Activate the hstore extension in this migration (it is create-if-not-exists, so we would be fine).

Reply:
With the repo management work requiring the task serializer in the plugin api I am pretty sure this next release will require pulpcore 3.23 which means we should always have the hstore extension available. Pulpcore migrations always run first right? If so we should be good.

Reply:
We _are_ depending on a later migration on pulpcore, so yes we _are_ good. My question was only about style, thinking: "Since we use the extension explicitly, maybe we should depend on it explicitly."

Reply:
Yeah, we need the pulpcore changes anyway, so no reason not to use HStore here.